packages/vite/src/node/server/searchRoot.ts

```diff
@@ -0,0 +1,50 @@
+import fs from 'fs'
+import { dirname } from 'path'
+import { join } from 'path'
+
+// https://github.com/vitejs/vite/issues/2820#issuecomment-812495079
+const ROOT_FILES = [
+  '.git',
+
+  // https://pnpm.js.org/workspaces/
+  'pnpm-workspace.yaml'
+
+  // https://rushjs.io/pages/advanced/config_files/
+  // 'rush.json',
+
+  // https://nx.dev/latest/react/getting-started/nx-setup
+  // 'workspace.json',
+  // 'nx.json'
```

Comment:
I am not sure if we should add `rush` and `nx` to the list, as even when the file presents, it does not necessary means it's workspace root, not?

Reply:
Maybe @pngwn could comment on this (https://github.com/vitejs/vite/issues/2820#issuecomment-812495079). But we can also add default support for `rush` and `nx` in future versions, I don't think they are strictly needed for this PR.